spec.html

```diff
@@ -22028,7 +22058,8 @@ <h1>
                 1. Return ? _status_.
               1. Else,
                 1. Assert: _iterationKind_ is ~iterate~.
-                1. Return ? IteratorClose(_iteratorRecord_, _status_).
+                1. Perform IteratorCloseForThrow(_iteratorRecord_).
```

Comment:
Is `_status_` necessarily a throw completion here?

Reply:
Yes: `_status_` here can only arise from DestructuringAssignmentEvaluation, BindingInitialization, Evaluation of `_lhs_` (which is either a LeftHandSideExpression, a ForBinding, a ForDeclaration, or a ForBinding; for each of these Evaluation to throw or normal completions), PutValue, ForDeclarationBindingInitialization, or InitializeReferencedBinding. And those all can only return a throw completion or a normal completion, so guarding on abruptness of `_status_ suffices.

Reply:
DestructuringAssignmentEvaluation, BindingInitialization, PutValue, ForDeclarationBindingInitialization, and InitializeReferencedBinding are all declared to return normal or *abrupt*. Can we change `abrupt` to `throw` for them?

And for the Evaluation of `_lhs_`, maybe when `_lhsRef_` is abrupt, there could be an `Assert` that it's a throw completion.

Reply:
(This reminded me of #2676 and https://github.com/tc39/ecma262/pull/2676#issuecomment-1058800656 in particular.)

Reply:
Ah jeez, you're right, I always forget about `yield`. These are wrong.